content/docs/atom-guide/dev-guide/shaders/azsl/unbounded-arrays.md

```diff
@@ -0,0 +1,306 @@
+    };
+```
+Now, let's review what will happen to all the examples above when using '--unique-idx', which is the case for Vulkan & Metal.
+
+### Example 6: Based on Example 1, with '--unique-idx'
+Same code as Example 1, but it is assumed to be compiled with '--unique-idx' (ON), which is required by Vulkan & Metal. Please note how the register index always increments, regardless of the resource type.  
+```cpp
+    ShaderResourceGroupSemantic slot1
+    {
+        FrequencyId = 1;
+    };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For Vulkan & Metal, --unique-idx is ON, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV1;      // Will be bound to t1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register).
+        Texture2D<float4>        m_texSRV2;      // Will be bound to t2 register.
+        RWTexture2D<float4>      m_texUAV1;      // Will be bound to u3 register.
+    	RWTexture2D<float4>      m_texUAV2;      // Will be bound to u4 register.
+        Sampler                  m_sampler1;     // Will be bound to s5 register.
+        Sampler                  m_sampler2;     // Will be bound to s6 register.
+        ConstantBuffer<MyStruct> m_cb1; // Will be bound to b7 register.
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b8 register.
+    };
+```
+### Example 7: Based on Example 2, with '--unique-idx'. Error Case.
+```cpp
+    ShaderResourceGroupSemantic slot1
+    {
+        FrequencyId = 1;
+    };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For Vulkan & Metal, --unique-idx is ON, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV_unbounded[]; // Will be bound to t1 register(Because The ConstantBuffer reserved for SRG1 will be bound to b0 register), and     will take over ALL1+. 
+        Texture2D<float4>        m_texSRV1;      // ERROR. There's no tN available to bind this resource to.
+        Texture2D<float4>        m_texSRV2;      // ERROR. There's no tN available to bind this resource to.
+        RWTexture2D<float4>      m_texUAV1;      // ERROR. There's no uN available to bind this resource to.
+    	RWTexture2D<float4>      m_texUAV2;      // ERROR. There's no uN available to bind this resource to.
+        Sampler                  m_sampler1;     // ERROR. There's no sN available to bind this resource to.
+        Sampler                  m_sampler2;     // ERROR. There's no sN available to bind this resource to.
+        ConstantBuffer<MyStruct> m_cb1; // ERROR. There's no bN available to bind this resource to.
+        ConstantBuffer<MyStruct> m_cb2; // ERROR. There's no bN available to bind this resource to.
+    };
```

Comment:
```suggestion
ShaderResourceGroupSemantic slot1
{
    FrequencyId = 1;
};

struct MyStruct
{
    float3 m_a;
    float3 m_b;
};

//For Vulkan & Metal, --unique-idx is ON, so the expected register assignment is as follows
// The ConstantBuffer reserved for SRG1 binds to b0 register.
ShaderResourceGroup SRG1 : slot1
{
    Texture2D<float4>        m_texSRV_unbounded[];  // Binds to t1 register, and  take over all 1+ registers.
    Texture2D<float4>        m_texSRV1;             // ERROR. There's no tN available to bind this resource to.
    Texture2D<float4>        m_texSRV2;             // ERROR. There's no tN available to bind this resource to.
    RWTexture2D<float4>      m_texUAV1;             // ERROR. There's no uN available to bind this resource to.
    RWTexture2D<float4>      m_texUAV2;             // ERROR. There's no uN available to bind this resource to.
    Sampler                  m_sampler1;            // ERROR. There's no sN available to bind this resource to.
    Sampler                  m_sampler2;            // ERROR. There's no sN available to bind this resource to.
    ConstantBuffer<MyStruct> m_cb1;                 // ERROR. There's no bN available to bind this resource to.
    ConstantBuffer<MyStruct> m_cb2;                 // ERROR. There's no bN available to bind this resource to.
};
```
* Moved comment: "(Because the ConstantBuffer reserved for….)" to simplify the comments for each resource declaration. 
* Remove indent. It's not necessary when defining within a code snippet (```cpp). This also allows longer comments to fit on the page.
* Simplified "will be bound to" to "binds to".
* Simplified "will take over" to "takes over"

Reply:
Fixed.